roles/aks/tasks/create_cluster.yaml

```diff
@@ -0,0 +1,85 @@
+---
+- name: Create Directory
+  file:
+    path: /tmp/aks
+    state: directory
+
+- name: Generate an OpenSSH Key Pair
+  community.crypto.openssh_keypair:
+    path: /tmp/aks/id_rsa
+
+# https://docs.ansible.com/ansible/latest/collections/azure/azcollection/azure_rm_aks_module.html
+- name: Create a managed Azure Kubernetes Services (AKS) instance
+  collections:
+    - azure.azcollection
+  azure_rm_aks:
+    name: "{{ aks.name }}"
+    kubernetes_version: 1.21.2
+    location: eastus
+    dns_prefix: "{{ aks.name }}"
+    enable_rbac: true
+    resource_group: "{{ aks.resource_group }}"
+    network_profile:
+      network_policy: calico
+    linux_profile:
+      admin_username: aks
+      ssh_key: "{{ lookup('file', '/tmp/aks/id_rsa.pub') }}" 
+    agent_pool_profiles:
+      - name: default
+        count: "{{ aks.node_count }}"
+        availability_zones:
+          - 1
+          - 2
+          - 3
+        vm_size: "{{ aks.node_size }}"
+        mode: System
+        enable_auto_scaling: false
+        type: VirtualMachineScaleSets
+
+- name: Create Kube Directory
+  file:
+    path: ~/.kube
+    state: directory
+
+- name: Get facts for Azure Kubernetes Service
+  collections:
+    - azure.azcollection
+  azure_rm_aks_info:
+    name: "{{ aks.name }}"
+    resource_group: "{{ aks.resource_group }}"
+    show_kubeconfig: admin
+  register: result
+
+- name: Set Kube Config
+  copy:
+    content: "{{ result.aks[0].kube_config }}"
+    dest: "~/.kube/config"
+
+- name: Create Backup Blob Container
```

Comment:
should we do this here or on the backup tasks?

Reply:
I'm on the fence here. I suppose it should probably move into the other role

Reply:
I remembered the reason I put this here. Basically right now de-provision leaves backups behind in the Velero branch I will PR next. However, I was thinking if we wanted to have a flag/etc to do a hard delete later on it would be easier if all of the AWS/AKS assets were managed through their repsective roles and not Veleros. I.e keep Velero role as simple as possible. It's why I put the Service Principals in the AKS/Kops roles here. They are used only for Velero (AWS one is in the Velero PR) atm, but could be used for other purposes as well.

Reply:
:thinking: but Velero is the only making use of the backup blob, no?